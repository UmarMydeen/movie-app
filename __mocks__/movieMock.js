const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster',
    imdbRating: '9.2',
    Poster: 'mockUrl'
}

export default movieMock;