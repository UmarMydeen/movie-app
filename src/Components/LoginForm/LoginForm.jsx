import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withRouter, Redirect } from 'react-router';
import validateLogin from '../../actions/validateLogin';

const LoginForm = ({
    isAuthenticated,
    actionValidateLogin,
    history,
}) => {
    const [userName, setUserName] = React.useState('');
    const [password, setPassword] = React.useState('');

    const inputOnchange = ({ target: {
        id: fieldId,
        value: fieldValue,
    } }) => {
        if (fieldId === 'email') {
            setUserName(fieldValue);
        }

        if (fieldId === 'password') {
            setPassword(fieldValue);
        }
    }

    const onFormSubmit = (event) => {
        event.preventDefault();
        actionValidateLogin({
            userName,
            password
        })
    }

    return (
        isAuthenticated ? <Redirect to="/movies" /> : (
            <div className="container w-25">
                <form onSubmit={onFormSubmit}>
                    <div className="mb-3">
                        <label for="email" className="form-label" > UserName </label >
                        <input
                            type="text"
                            className="form-control"
                            id="email"
                            aria-describedby="emailHelp"
                            onChange={inputOnchange}
                        />
                    </div >
                    <div className="mb-3">
                        <label for="password" className="form-label">Password</label>
                        <input
                            type="password"
                            className="form-control"
                            id="password"
                            onChange={inputOnchange}
                        />
                    </div>
                    <button type="submit" className="btn btn-primary" data-testid='loginFormBtn'>Submit</button>
                </form >
            </div >
        )
    )
}

LoginForm.propTypes = {
    actionValidateLogin: PropTypes.func.isRequired,
    isAuthenticated: PropTypes.bool.isRequired,
    history: PropTypes.shape({})
}

const mapStateToProps = (state) => {
    return {
        isAuthenticated: state.login.isAuth
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        actionValidateLogin: (form) => dispatch(validateLogin(form))
    }
};

export { LoginForm as TestableLoginForm };

const hocChain = compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps),
);

export default hocChain(LoginForm);