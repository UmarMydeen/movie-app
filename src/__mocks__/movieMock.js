const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster'
}

export default movieMock;