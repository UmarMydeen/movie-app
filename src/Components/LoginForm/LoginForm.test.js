import { fireEvent, render, screen } from '@testing-library/react';
import { TestableLoginForm as LoginForm } from './LoginForm';
import movieMock from '../../../__mocks__/movieMock';
import { BrowserRouter } from 'react-router-dom';

let mockProps;

beforeEach(() => {
    mockProps = {
        isAuthenticated: false,
        actionValidateLogin: jest.fn(),
        history: {
            push: jest.fn()
        },
    }
})

test('should load login form on initial render', () => {
    const component = render(<LoginForm {...mockProps} />);
    expect(component).toMatchSnapshot();
});

test('should redirect to dashboard when login successfull', () => {
    mockProps.isAuthenticated = true;
    const component = render(
        <BrowserRouter>
            <LoginForm {...mockProps} />
        </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
});

test('should call actionValidateLogin on form submit', () => {
    const component = render(<LoginForm {...mockProps} />);
    fireEvent.click(component.getByTestId('loginFormBtn'));
    expect(mockProps.actionValidateLogin).toHaveBeenCalled();
});