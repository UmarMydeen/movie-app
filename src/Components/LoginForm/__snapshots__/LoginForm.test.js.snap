// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load login form on initial render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container w-25"
      >
        <form>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
               UserName 
            </label>
            <input
              aria-describedby="emailHelp"
              class="form-control"
              id="email"
              type="text"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              id="password"
              type="password"
            />
          </div>
          <button
            class="btn btn-primary"
            data-testid="loginFormBtn"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container w-25"
    >
      <form>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="email"
          >
             UserName 
          </label>
          <input
            aria-describedby="emailHelp"
            class="form-control"
            id="email"
            type="text"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="password"
          >
            Password
          </label>
          <input
            class="form-control"
            id="password"
            type="password"
          />
        </div>
        <button
          class="btn btn-primary"
          data-testid="loginFormBtn"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should redirect to dashboard when login successfull 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
